language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/MappingContextProvider.java

```diff
@@ -187,7 +194,99 @@ protected boolean hasReadonlyKW(Node evalNodeAtCursor) {
+    }
+
+    private List<LSCompletionItem> getSpreadFieldCompletionItems(Node evalNode, BallerinaCompletionContext context) {
+        ContextTypeResolver typeResolver = new ContextTypeResolver(context);
+        Optional<TypeSymbol> resolvedType = evalNode.apply(typeResolver);
+
+        if (resolvedType.isEmpty() || resolvedType.get().typeKind() != TypeDescKind.MAP) {
+            return Collections.emptyList();
+        }
+        Predicate<Symbol> symbolFilter = this.getVariableFilter().or(symbol -> (symbol.kind() == FUNCTION));
+        List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition()).stream()
+                .filter(symbolFilter.and(symbol -> {
+                    Optional<TypeSymbol> typeDescriptor = SymbolUtil.getTypeDescriptor(symbol);
+                    return typeDescriptor.isPresent() && typeDescriptor.get().subtypeOf(resolvedType.get())
+                            && (CommonUtil.getRawType(typeDescriptor.get()).typeKind() == TypeDescKind.MAP
+                            || CommonUtil.getRawType(typeDescriptor.get()).typeKind() != TypeDescKind.RECORD);
+                })).collect(Collectors.toList());
+
+        return getSpreadFieldCompletionItemList(visibleSymbols, context);
+    }
+
+    private List<LSCompletionItem> getSpreadFieldCompletionItems(BallerinaCompletionContext context,
+                                                                 List<RecordFieldSymbol> validFields) {
+        Predicate<Symbol> symbolFilter = this.getVariableFilter().or(symbol -> (symbol.kind() == FUNCTION));
+        List<Symbol> visibleSymbols = context.visibleSymbols(context.getCursorPosition()).stream()
+                .filter(symbolFilter.and(symbol -> isSpreadable(symbol, validFields)))
+                .collect(Collectors.toList());
+        return this.getSpreadFieldCompletionItemList(visibleSymbols, context);
+    }
+
+    private boolean isSpreadable(Symbol symbol, List<RecordFieldSymbol> fields) {
+        Optional<TypeSymbol> typeDescriptor = SymbolUtil.getTypeDescriptor(symbol);
+        if (typeDescriptor.isEmpty()) {
+            return false;
         }
+        TypeSymbol rawType;
+        if (typeDescriptor.get().typeKind() == TypeDescKind.FUNCTION) {
+            Optional<TypeSymbol> returnTypeSymbol =
+                    ((FunctionTypeSymbol) typeDescriptor.get()).returnTypeDescriptor();
+            if (returnTypeSymbol.isEmpty()) {
+                return false;
+            }
+            typeDescriptor = returnTypeSymbol;
+        }
+
+        rawType = CommonUtil.getRawType(typeDescriptor.get());
+        if (rawType.typeKind() != TypeDescKind.RECORD) {
+            return false;
+        }
+        Map<String, RecordFieldSymbol> recordFields = ((RecordTypeSymbol) rawType).fieldDescriptors();
+        for (RecordFieldSymbol fieldSymbol : recordFields.values()) {
+            if (!isAMember(fieldSymbol, fields)) {
+                return false;
+            }
+        }
```

Comment:
Shall we use the stream API here? Maybe we can go with `anyMatch`? So we can have single return statement.